akka-stream/src/main/scala/akka/stream/impl/io/IOSinks.scala

```diff
@@ -66,3 +74,163 @@ private[akka] final class OutputStreamSink(createOutput: () ⇒ OutputStream, va
   override def withAttributes(attr: Attributes): Module =
     new OutputStreamSink(createOutput, attr, amendShape(attr))
 }
+
+/**
+ * INTERNAL API
+ * Creates simple synchronous (Java 6 compatible) Sink which provide all incoming element on read request via
+ * InputStream.
+ */
+private[akka] class InputStreamSink(val timeout: FiniteDuration,
+                                    val attributes: Attributes, shape: SinkShape[ByteString])
+  extends SinkModule[ByteString, (InputStream, Future[Long])](shape) {
+
+  protected def getSubscriber(sharedBuffer: BlockingQueue[ByteString]) =
```

Comment:
Will fix this. Please don't review InputStreamSink/InputStreamSubscriber part as this is still actor based (previous version). Now I'm trying to implement with using GraphStage as @drewhk  suggested